abis_mapping/base/mapper.py

```diff
@@ -311,17 +311,16 @@ def schema(
         return s_class.model_dump(exclude_none=discard_optional)
 
     @final
-    @classmethod
     @property
-    def fields(cls) -> dict[str, types.schema.Field]:
+    def fields(self) -> dict[str, types.schema.Field]:
```

Comment:
Using `@classmethod` and `@property` together is deprecated, and hence mypy or flake8 was complaining about this, so I had to remove one of them.

It was a smaller change to make it an instance property, compared to removing `@property`, which would've meant adding `()` everywhere it is used.

Is there a reason to change the others, other than consistency?
Note its ok for this instance method to call classmethods like `schema()`

Reply:
Also I think making `schema()` an instance method, isn't a good idea, because of its `@functools.lru_cache` - the cache would effectively become per-instance, and less effective.
Same thing for `metadata()`